concent_api/core/views.py

```diff
@@ -1241,6 +1209,712 @@ def store_message_and_message_status(
+        report_computed_task = deserialize_message(database_message.subtask.report_computed_task.data.tobytes())
+        mark_message_as_delivered(database_message)
+        response_to_client = message.concents.ForceGetTaskResultRejected(
+            force_get_task_result = message.concents.ForceGetTaskResult(
+                report_computed_task = report_computed_task,
+            )
+        )
+        logging.log_receive_message_from_database(
+            response_to_client,
+            database_message.client.public_key,
+            database_message.response_type,
+            database_message.queue
+        )
+        return response_to_client
+
+    elif database_message.response_type == PendingResponse.ResponseType.ForceGetTaskResultFailed.name:  # pylint: disable=no-member
+        task_to_compute = deserialize_message(database_message.subtask.task_to_compute.data.tobytes())
+        mark_message_as_delivered(database_message)
+        response_to_client = message.concents.ForceGetTaskResultFailed(
+            task_to_compute = task_to_compute,
+        )
+        logging.log_receive_message_from_database(
+            response_to_client,
+            database_message.client.public_key,
+            database_message.response_type,
+            database_message.queue
+        )
+        return response_to_client
+
+    elif (database_message.response_type == PendingResponse.ResponseType.ForceGetTaskResultUpload.name or  # pylint: disable=no-member
+            database_message.response_type == PendingResponse.ResponseType.ForceGetTaskResult.name):  # pylint: disable=no-member
+        force_get_task_result   = deserialize_message(database_message.subtask.force_get_task_result.data.tobytes())
+        file_transfer_token     = create_file_transfer_token(
+            force_get_task_result.report_computed_task,
+            client_public_key,
+        )
+        if database_message.subtask.provider.public_key_bytes == client_public_key:
+            file_transfer_token.operation = 'upload'
+        elif database_message.subtask.requestor.public_key_bytes == client_public_key:
+            file_transfer_token.operation = 'download'
+        else:
+            return None
+
+        mark_message_as_delivered(database_message)
+        response_to_client = message.concents.ForceGetTaskResultUpload(
+            file_transfer_token     = file_transfer_token,
+            force_get_task_result   = force_get_task_result,
+        )
+        logging.log_receive_message_from_database(
+            response_to_client,
+            database_message.client.public_key,
+            database_message.response_type,
+            database_message.queue
+        )
+        return response_to_client
+
+    elif database_message.response_type == PendingResponse.ResponseType.ForgetGetTaskResultDownload.name:  # pylint: disable=no-member
+        raise NotImplementedError
+        # TODO Message isn't even implemented yet in golem_messages library
```

Comment:
This needs to be fixed before the release. Otherwise we cannot deliver a token to the requestor.

If this message is missing, it needs to be reported to Golem.

Reply:
What about this? Is the message missing? Has it been reported?

Reply:
Already implemented and added to Concent.